### PR TITLE
Advanced: chapter from 1 to 3

### PR DESCRIPTION
**Pull request by Vretamoza**

### CODE REVIEW

File: kv/lib/kv/registry.ex [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  """
+  def lookup(server, name) do
+    GenServer.call(server, {:lookup, name})
+  end
+
+  @doc """
+  Ensures there is a bucket associated with the given `name` in `server`.
+  """
+  def create(server, name) do
+    GenServer.cast(server, {:create, name})
```
Comment (juanhurtado10) [inline kv/lib/kv/registry.ex:68]: @Vretamoza what does the `Genserver.cast/2` do?

File: kv/lib/kv/registry.ex [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    GenServer.start_link(__MODULE__, :ok, opts)
+  end
+
+  @doc """
+  Looks up the bucket pid for `name` stored in `server`.
+
+  Returns `{:ok, pid}` if the bucket exists, `:error` otherwise.
+  """
+  def lookup(server, name) do
+    GenServer.call(server, {:lookup, name})
```
Comment (juanhurtado10) [inline kv/lib/kv/registry.ex:61]: @Vretamoza what is the difference between `GenServer.call` and `GenServer.cast` ?

File: kv/lib/kv/registry.ex [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    else
+      {:ok, bucket} = KV.Bucket.start_link([])
+      ref = Process.monitor(bucket)
+      refs = Map.put(refs, ref, name)
+      names = Map.put(names, name, bucket)
+      {:noreply, {names, refs}}
+    end
+  end
+
+  @impl true
```
Comment (juanhurtado10) [inline kv/lib/kv/registry.ex:34]: @Vretamoza what does this mean?

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: kv/lib/kv/registry.ex (hunk 1 above)
Comment (Vretamoza) [reply kv/lib/kv/registry.ex:68]: The GenServer.cast makes a request to the server but it doesn't expect a response from it, So it doesn't have to wait for a response. It receives two arguments: the request and the current state of the GenServer. In this case it creates the bucket with the given name in the params.

File: kv/lib/kv/registry.ex (hunk 2 above)
Comment (Vretamoza) [reply kv/lib/kv/registry.ex:61]: The principal difference between GenServer.call and GenServer.cast is that the first one is synchronous, so the client sends a request to the server and waits for a response from it. The second one is asynchronous, so it just sends the request but doesn't wait for an answer.

File: kv/lib/kv/registry.ex (hunk 3 above)
Comment (Vretamoza) [reply kv/lib/kv/registry.ex:34]: The @impl true means that the function that comes next is intended to be a GenServer callback.

**Pull request merged**